Mathlib/Data/Matrix/Defs.lean

```diff
@@ -565,3 +566,28 @@ lemma col_transpose (A : Matrix m n α) : Aᵀ.col = A.row := rfl
 end RowCol
 
 end Matrix
+
+namespace Set
+
+/-- Given a set `S`, `S.matrix` is the set of matrices `M`
+all of whose entries `M i j` belong to `S`. -/
+def matrix (S : Set α) : Set (Matrix m n α) := {M | ∀ i j, M i j ∈ S}
+
+theorem mem_matrix {S : Set α} {M : Matrix m n α} :
+    M ∈ S.matrix ↔ ∀ i j, M i j ∈ S := .rfl
+
+end Set
+
+namespace Matrix
+
+variable {S : Set α}
+
+@[simp]
+theorem transpose_mem_matrix_iff {M : Matrix m n α} :
+    Mᵀ ∈ S.matrix ↔ M ∈ S.matrix := by
+  simp only [Set.mem_matrix, transpose_apply]; exact forall_comm
+
+theorem submatrix_mem_matrix {M : Matrix m n α} (hM : M ∈ S.matrix) {r : l → m} {c : o → n} :
```

Comment:
Is this true as an `Iff` too?

Reply:
The version you haven doesn't look very useful to me. I think we want one with `r` and `c` surjective instead, where I think the `Iff` is true.